bin/common.ml

```diff
@@ -684,6 +690,12 @@ let term =
   let build_dir = Option.value ~default:default_build_dir build_dir in
   let root = Workspace_root.create ~specified_by_user:root in
   let config = config_of_file config_file in
+  let rpc =
+    if rpc || watch then
```

Comment:
In which case would we want a Dune not started in watch mode to serve RPC request?

(except for RPC requests coming from command started by Dune, but that's another story)

Reply:
> In which case would we want a Dune not started in watch mode to serve RPC request?

So that we can do something like `$ dune build --rpc ..`. This will build something and allow rpc connections in the meanwhile. It would also be nice to have it with `$ dune exec --rpc` for the same reason, but it's a little harder because we are using `exec`.

Reply:
For `dune exec`, it seems reasonable to me to stop the server right before doing the `exec`.

For `dune build --rpc ..`, I admit that I find that a bit dubious. In particular, it is not clear to me when `dune build --rpc` should stop: is it when the build has terminated, when all the current requests have been answered, or something else? Finally if a program starts `dune build --rpc` and then tries to issue a rpc request, then there is a race condition; dune could finish before the program has had a chance to issue the rpc request.

Reply:
The conclusion was just to drop all this `--rpc` business.